File: Seccion_II_preprocessing/missing_data.R
#Plantilla de pre procesado de datos -- Datos faltantes

# Importamos el Dataset
dataset = read.csv("Data.csv")

#Tratamiento de los valores NA -Tomamos primero la columna edad
#y calculamos la media, luego hacemos lo mismo con la columna salario

dataset$Age = ifelse(is.na(dataset$Age), 
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     dataset$Age)


dataset$Salary = ifelse(is.na(dataset$Salary), 
                        ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        dataset$Salary)

File: Seccion_II_preprocessing/missing_data.py

#Plantilla de pre procesado -- Datos faltantes
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd 

#Importamos el dataset
dataset = pd.read_csv('Data.csv')
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:,-1].values

#Tratamiento de los valores NaN
from sklearn.impute import SimpleImputer
imputer = SimpleImputer(missing_values = np.nan, strategy = "mean")
Imputer = imputer.fit(X[:, 1:3])
X[:, 1:3] = imputer.transform(X[:,1:3])
